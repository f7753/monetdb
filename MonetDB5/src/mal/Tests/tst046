#testing flt/lng/int coercions

b:=new(int,int);
insert(b,1,1);
insert(b,2,4);
insert(b,3,9);
print(b);

f:= flt(b);
print(f);

l:=lng(b);
print(l);

i:= int(f);
print(i);
