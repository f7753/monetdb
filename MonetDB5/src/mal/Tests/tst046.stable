# Monet Database Server V5.0
# Copyright (c) 1993-2004, CWI. All rights reserved.
# compiled for 32biti686-pc-linux-gnu/32bit.
# config:/ufs/mk/monet5/src/mal/Tests/tst.conf
# dbfarm:/ufs/mk/monet5/Linux/var/MonetDB
# dbname:demo
##testing flt/lng/int coercions
#
#b:=new(:int,:int);
#insert(b,1,1);
#insert(b,2,4);
#insert(b,3,9);
#print(b);
#
#f:= flt(b);
#print(f);
#
#l:=lng(b);
#print(l);
#
#i:= int(f);
#print(i);
#
function Admin.main():int;	# 0  (main:int)
    b := bbp.new(:int,:int);	# 1 CMDBATnew (b:bat[int,int])<-($2:int)($3:int)
    bat.insert(b,1,1);	# 2 CMDinsert_bun ($4:void)<-(b:bat[int,int])($5:int)($5:int)
    bat.insert(b,2,4);	# 3 CMDinsert_bun ($6:void)<-(b:bat[int,int])($7:int)($8:int)
    bat.insert(b,3,9);	# 4 CMDinsert_bun ($9:void)<-(b:bat[int,int])($10:int)($11:int)
    io.print(b);	# 5 CMDprint_val ($12:int)<-(b:bat[int,int])
    f := batcalc.flt(b);	# 6 CMDconvertint_flt (f:bat[int,flt])<-(b:bat[int,int])
    io.print(f);	# 7 CMDprint_val ($14:int)<-(f:bat[int,flt])
    l := batcalc.lng(b);	# 8 CMDconvertint_lng (l:bat[int,lng])<-(b:bat[int,int])
    io.print(l);	# 9 CMDprint_val ($16:int)<-(l:bat[int,lng])
    i := batcalc.int(f);	# 10 CMDconvertflt_int (i:bat[int,int])<-(f:bat[int,flt])
    io.print(i);	# 11 CMDprint_val ($18:int)<-(i:bat[int,int])
end main;	# 12  
#-----------------#
# h	tmp_4	  # name
# int	int	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_5	  # name
# int	flt	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_6	  # name
# int	lng	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_7	  # name
# int	int	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
>