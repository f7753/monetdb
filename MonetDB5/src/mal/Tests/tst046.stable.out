stdout of test 'tst046` in directory 'src/mal` itself:


# 07:58:45 >  
# 07:58:45 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/Linux/etc/MonetDB5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5/Linux/var/MonetDB5/log" --set mapi_port=42158 --set sql_port=55708 --set monet_prompt= --trace "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --dbname=mTests_src_mal  tst046.mal </dev/null
# 07:58:45 >  

# Monet Database Server V4.99.19
# Copyright (c) 1993-2004, CWI. All rights reserved.
# Compiled for i686-redhat-linux-gnu/32bit; dynamically linked.
# config:/ufs/mk/monet5/Linux/etc/MonetDB5.conf
# dbfarm:/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm
# dbname:mTests_src_mal
# Visit http://monetdb.cwi.nl/ for further information.
##testing flt/lng/int coercions
#
#b:=new(:int,:int);
#insert(b,1,1);
#insert(b,2,4);
#insert(b,3,9);
#print(b);
#
#f:= flt(b);
#print(f);
#
#l:=lng(b);
#print(l);
#
#i:= int(f);
#print(i);
#
function user.main():int;	# 0  (main:int)
    b := bbp.new(:int,:int);	# 1 CMDBATnew (b:bat[:int,:int])<-($2:int)($3:int)
    bat.insert(b,1,1);	# 2 BKCinsert_bun ($4:bat[:int,:int])<-(b:bat[:int,:int])($5:int)($5:int)
    bat.insert(b,2,4);	# 3 BKCinsert_bun ($6:bat[:int,:int])<-(b:bat[:int,:int])($7:int)($8:int)
    bat.insert(b,3,9);	# 4 BKCinsert_bun ($9:bat[:int,:int])<-(b:bat[:int,:int])($10:int)($11:int)
    io.print(b);	# 5 IOprint_val ($12:int)<-(b:bat[:int,:int])
    f := batcalc.flt(b);	# 6 CMDconvertint_flt (f:bat[:int,:flt])<-(b:bat[:int,:int])
    io.print(f);	# 7 IOprint_val ($14:int)<-(f:bat[:int,:flt])
    l := batcalc.lng(b);	# 8 CMDconvertint_lng (l:bat[:int,:lng])<-(b:bat[:int,:int])
    io.print(l);	# 9 IOprint_val ($16:int)<-(l:bat[:int,:lng])
    i := batcalc.int(f);	# 10 CMDconvertflt_int (i:bat[:int,:int])<-(f:bat[:int,:flt])
    io.print(i);	# 11 IOprint_val ($18:int)<-(i:bat[:int,:int])
end main;	# 12  
#-----------------#
# h	tmp_4	  # name
# int	int	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_5	  # name
# int	flt	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_6	  # name
# int	lng	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]
#-----------------#
# h	tmp_7	  # name
# int	int	  # type
#-----------------#
[ 1,	  1	  ]
[ 2,	  4	  ]
[ 3,	  9	  ]

# 07:58:45 >  
# 07:58:45 >  Done.
# 07:58:45 >  

